### PR TITLE
add gitlab merge request bitbar plugin

### PR DESCRIPTION
**Pull request by SimeonC**
Adding a new plugin to allow display the status of currently assigned and reported GitLab Merge Requests.

### CODE REVIEW

**Comment by SimeonC** [comment]
So it looks like the CI is failing due to python@2 being removed from homebrew, not sure what I can do to fix this 😅 https://github.com/Homebrew/homebrew-core/pull/49796

**Comment by tresni** [comment]
Toggling PR to trigger Travis build

File: Dev/Gitlab/gitlab_merge_requests.js [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+            if (Buffer.isBuffer(chunk)) {
+              responseBufs.push(chunk);
+            } else {
+              responseStr = responseStr + chunk;
+            }
+          })
+          .on('end', () => {
+            responseStr =
+              responseBufs.length > 0
+                ? Buffer.concat(responseBufs).toString(responseEncoding)
```
Comment (tresni) [inline Dev/Gitlab/gitlab_merge_requests.js:132]: line 132, col 17, Misleading line break before '?'; readers may interpret this as an expression boundary.

File: Dev/Gitlab/gitlab_merge_requests.js [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        reject(error);
+      });
+    request.write('');
+    request.end();
+  });
+}
+
+const cachedProjectNames = {};
+
+async function getMRs() {
```
Comment (tresni) [inline Dev/Gitlab/gitlab_merge_requests.js:149]: line 149, col 1, 'async functions' is only available in ES8 (use 'esversion: 8').

File: Dev/Gitlab/gitlab_merge_requests.js [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+  const statusCount = {};
+  function incrementStatusCount(key) {
+    statusCount[key] = statusCount[key] || 0;
+    statusCount[key] += 1;
+  }
+
+  const mergeRequestsByProjectId = {};
+
+  await Promise.all(
+    allMRs.map(async (mergeRequest) => {
```
Comment (tresni) [inline Dev/Gitlab/gitlab_merge_requests.js:203]: line 203, col 16, 'async functions' is only available in ES8 (use 'esversion: 8').

File: Dev/Gitlab/gitlab_merge_requests.js [hunk 4]
```diff
@@ -0,0 +1,433 @@
+#!/usr/bin/env /usr/local/bin/node
+// jshint asi:true
```
Comment (tresni) [inline Dev/Gitlab/gitlab_merge_requests.js:2]: Add ` esversion: 8` to resolve 2 of the issues below

**Comment by SimeonC** [comment]
Sorry for taking so long, had a bunch of work to do. Made some minor updates as well as I use this one heavily 😆 

I'll toggle the PR to get travis to run, I should have got all the issues this time.

**Pull request merged**